src/Tasks/Microsoft.Common.CrossTargeting.targets

```diff
@@ -227,6 +227,13 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     build.
   -->
   <PropertyGroup>
+    <!--
+        Don't import project extensions during restore because NuGet restore generates them.  Importing them during restore will embed
+        the pre-restore files in the binary log and then NuGet won't be able to embed the generated one after restore.  If some other
+        project extension mechanism wants to import project extensions during restore, they need to explicitly set ImportProjectExtensionTargets
+    -->
+    <ImportProjectExtensionTargets Condition="'$(ImportProjectExtensionTargets)' == '' And '$(MSBuildIsRestoring)' == 'true'">false</ImportProjectExtensionTargets>
```

Comment:
Should I put the `MSBuildIsRestoring` property behind a change wave to accomplish this?  Or is there a change wave property/mechanism for props/targets?

Reply:
There is a property function (`$([MSBuild]::AreFeaturesEnabled('17.10'))`) and I think I'd rather use that since _adding_ the property should be innocuous. But if you've got a reason to go the other way I'd be fine with that too.